datafusion/optimizer/src/replace_distinct_aggregate.rs

```diff
@@ -82,10 +82,11 @@ impl OptimizerRule for ReplaceDistinctWithAggregate {
                 for dep in input.schema().functional_dependencies().iter() {
                     // If distinct is exactly the same with a previous GROUP BY, we can
                     // simply remove it:
-                    if dep.source_indices[..field_count]
-                        .iter()
-                        .enumerate()
-                        .all(|(idx, f_idx)| idx == *f_idx)
+                    if dep.source_indices.len() >= field_count
```

Comment:
is this change required for the test to pass? Or is it a "drive by" cleanup (made while you were making the change already but not required)?

Reply:
This one was required to make the test pass

Reply:
Nice -- so this also fixes a bug 🚀